test/MatrixSets/MatrixZonotope.jl

```diff
@@ -120,9 +121,41 @@ for N in @tN([Float64, Float32, Rational{Int}])
     @test expMZ.M == A_mz
     @test size(expMZ) == size(A_mz)
     @test_throws AssertionError MatrixZonotopeExp(B_mz)
+
+    # vectorize and matrixize
+    c = N[1 0; 0 3]
+    gens = [N[1 -1; 0 2]]
+    MZ = MatrixZonotope(c, gens)
+    Z = vectorize(MZ)
+    @test Z == Zonotope(N[1, 0, 0, 3], hcat(N[1, 0, -1, 2]))
+    MZ2 = matrixize(Z, (2, 2))
+    @test isapprox(center(MZ), center(MZ2))
+    @test isapprox(generators(MZ), generators(MZ2))
+
+    # order
+    @test order(MZ) == 1//4 && order(MZ) == order(Z)
+
+    # remove redundant generators 
+    MZ2 = MatrixZonotope(c, [N[1 4; 0 -2], N[-1 1; 0 -1], N[1 -1; 0 1]])
+    MZred = remove_redundant_generators(MZ2)
+    @test ngens(MZred) == 2
+    @static if isdefined(@__MODULE__, :Polyhedra)
+        @test isequivalent(vectorize(MZ2), vectorize(MZred)) == true
```

Comment:
Let's do some remote debugging:
```suggestion
        @show vectorize(MZ2)
        @show vectorize(MZred)
        @test isequivalent(vectorize(MZ2), vectorize(MZred)) == true
```

Reply:
I think it has something to do with `matrixize` and `vectorize`. 
Before I had those tests on line 128
```julia
MZ = MatrixZonotope(c, gens)
    Z = vectorize(MZ)
    @test Z == Zonotope(N[1, 0, 0, 3], hcat(N[1, 0, -1, 2]))
    MZ2 = matrixize(Z, (2, 2))
    @test MZ == MZ2
```
But they failed even though `MZ` and `MZ2` look identical when printed out.
I had to change to 
```julia
MZ = MatrixZonotope(c, gens)
    Z = vectorize(MZ)
    @test Z == Zonotope(N[1, 0, 0, 3], hcat(N[1, 0, -1, 2]))
    MZ2 = matrixize(Z, (2, 2))
    @test isapprox(center(MZ), center(MZ2))
    @test isapprox(generators(MZ), generators(MZ2))
```
for it to work

Reply:
```julia
vectorize(MZ2) = Zonotope{Float32, Vector{Float32}, Matrix{Float32}}(Float32[1.0, 0.0, 0.0, 3.0], Float32[1.0 -1.0 1.0; 0.0 0.0 0.0; 4.0 1.0 -1.0; -2.0 -1.0 1.0])
vectorize(MZred) = Zonotope{Float32, Vector{Float32}, Matrix{Float32}}(Float32[1.0, 0.0, 0.0, 3.0], Float32[1.0 1.9999999; 0.0 -0.0; 4.0 -1.9999999; -2.0 1.9999999])
```
So the result seems to be rounded. Very interesting because it does not happen for me, and I use the same Julia version.
So I guess the `isequivalent` implementation is not robust to minor imprecision. In that case you should just check something simpler and add a note so we do not forget why it is like that (also to the other place you mentioned, please).

Reply:
Actually, for `Float64` you can keep the test!

Reply:
> also to the other place you mentioned, please

The other place is fine actually. I just realized that `==` is not overloaded to check equality of `MatrizZonotope`s, and there is no rounding. 

I added the comments for the `isequivalent` case instead